L1Trigger/L1THGCal/plugins/veryfrontend/HGCalVFEProcessorSums.cc

```diff
@@ -0,0 +1,100 @@
+#include "L1Trigger/L1THGCal/interface/veryfrontend/HGCalVFEProcessorSums.h"
+#include <limits>
+
+#include "DataFormats/Candidate/interface/LeafCandidate.h"
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+
+DEFINE_EDM_PLUGIN(HGCalVFEProcessorBaseFactory, 
+        HGCalVFEProcessorSums,
+        "HGCalVFEProcessorSums");
+
+
+HGCalVFEProcessorSums::
+HGCalVFEProcessorSums(const edm::ParameterSet& conf) : HGCalVFEProcessorBase(conf),
+  vfeLinearizationImpl_(conf),
+  vfeSummationImpl_(conf),
+  calibration_( conf.getParameterSet("calib_parameters") )
+{ 
+}
+
+void
+HGCalVFEProcessorSums::run(const HGCalDigiCollection& ee,
+                           const HGCalDigiCollection& fh, 
+                           const HGCalDigiCollection& bh, 
+                           l1t::HGCalTriggerCellBxCollection& triggerCellColl, 
+                           const edm::EventSetup& es) 
+{ 
+  calibration_.eventSetup(es);
+
+  std::vector<HGCalDataFrame> dataframes;
+  std::vector<std::pair<DetId, uint32_t >> linearized_dataframes;
+  std::map<HGCalDetId, uint32_t> payload;
+
+  // convert ee and fh hit collections into the same object  
+  if(!ee.empty())
+  {
+    for(const auto& eedata : ee)
+    { 
+      uint32_t module = geometry_->getModuleFromCell(eedata.id());
+      if(geometry_->disconnectedModule(module)) continue;
+      dataframes.emplace_back(eedata.id());
+      for(int i=0; i<eedata.size(); i++)
+      {
+        dataframes.back().setSample(i, eedata.sample(i));
+      }
+    }
+  }
+  else if(!fh.empty())
+  {
+    for(const auto& fhdata : fh)
+    {
+       uint32_t module = geometry_->getModuleFromCell(fhdata.id());
+       if(geometry_->disconnectedModule(module)) continue;
+       dataframes.emplace_back(fhdata.id());
+       for(int i=0; i<fhdata.size(); i++)
+       {
+         dataframes.back().setSample(i, fhdata.sample(i));
+       }
+     }
+   }
+  else if(!bh.empty())
+  {  
+     for(const auto& bhdata : bh)
+     { 
+       if(HcalDetId(bhdata.id()).subdetId()!=HcalEndcap) continue;
```

Comment:
appears not to be ready for HGCal v9 - a step backwards? Please describe the timescale for this new implementation to work with the new geometry

Reply:
Right, thanks, I missed that. Will be fixed.